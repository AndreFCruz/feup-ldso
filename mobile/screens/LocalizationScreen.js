import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'native-base';
import { connect } from 'react-redux';
import { getLocalization } from '../reducers/modules/facultyReducer';

class LocalizationScreen extends React.Component {

    componentDidMount() {
        this.props.getLocalization(this.props.name);
    }

    render() {
        const { name, loading } = this.props;

        if (loading) {
            return (
                <View>
                    <Text>
            Loading...
                    </Text>
                </View>
            );
        }
        return (
            <View>
                <Text>
          Placeholder for Localization Screen of {name}
                </Text>
            </View>
        );
    }
}

LocalizationScreen.propTypes = {
    name: PropTypes.string,
    loading: PropTypes.bool,
    getLocalization: PropTypes.func
};

const mapStateToProps = ({ faculty }) => ({
    name: faculty.name,
    loading: faculty.loading,
    localization: faculty.localization
});

const mapDispatchToProps = {
    getLocalization
};

export default connect(mapStateToProps, mapDispatchToProps)(LocalizationScreen);