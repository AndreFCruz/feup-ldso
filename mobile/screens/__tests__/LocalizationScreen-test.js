import 'react-native';
import React from 'react';
import LocalizationScreen from '../LocalizationScreen';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(loading = false) {

    const initialState = { 
        faculty: {
            name: 'feup',
            loading: loading,
            localization: 'test'
        },
        language: {
            selection: 'en'
        }
    };

    const wrapper = shallow(<LocalizationScreen />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: true
        });

    return wrapper;
}

it('renders loading screen', () => {
    const wrapper = setup(true);

    expect(wrapper.dive()).toMatchSnapshot();
});
    
it('renders screen placeholder content', () => {
    const wrapper = setup();

    expect(wrapper.dive()).toMatchSnapshot();
});

