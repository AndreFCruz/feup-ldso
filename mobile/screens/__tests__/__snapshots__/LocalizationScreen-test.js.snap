// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading screen 1`] = `
<Styled(ViewNB)>
  <Styled(Text)>
    Loading...
  </Styled(Text)>
</Styled(ViewNB)>
`;

exports[`renders screen placeholder content 1`] = `
<Styled(ViewNB)>
  <Styled(Text)>
    Placeholder for Localization Screen of 
    feup
  </Styled(Text)>
</Styled(ViewNB)>
`;
